aws/table_aws_sfn_state_machine_execution.go

```diff
@@ -131,18 +129,15 @@ func listStepFunctionsStateMachineExecutions(ctx context.Context, d *plugin.Quer
 
 	stateMachineArn := h.Item.(types.StateMachineListItem).StateMachineArn
 
-	equalQuals := d.EqualsQuals
 	// Minimize the API call with the given state machine ARN
-	if equalQuals["state_machine_arn"] != nil {
-		if equalQuals["state_machine_arn"].GetStringValue() != "" {
-			if equalQuals["state_machine_arn"].GetStringValue() != "" && equalQuals["state_machine_arn"].GetStringValue() != *stateMachineArn {
-				return nil, nil
-			}
-		} else if len(getListValues(equalQuals["state_machine_arn"].GetListValue())) > 0 {
-			if !strings.Contains(fmt.Sprint(getListValues(equalQuals["state_machine_arn"].GetListValue())), *stateMachineArn) {
-				return nil, nil
-			}
+	stateMachineArnQuals := getQualsValueByColumn(d.Quals, "state_machine_arn", "string")
```

Comment:
Were there any functional differences with this change, or more stylistic changes?

Reply:
As it uses `d.Quals` instead of `d.EqualsQuals`, I believe it also makes queries not using `=` but `in` more performant:
```
select * from aws_sfn_state_machine_execution
where state_machine_arn in (
  select state_machine_arn
  from aws_sfn_state_machine
  where name = 'mystatemachine'
)
```

See
* https://github.com/turbot/steampipe-plugin-sdk/blob/2b7903e9b3d0aee3b456819a8896d1fd4d4c73ce/plugin/query_data.go#L393
* https://github.com/turbot/steampipe-plugin-sdk/blob/49556d376edd992a9ff0cb60485bd50c098604ac/plugin/key_column_qual_map.go#L22
* https://github.com/turbot/steampipe-plugin-sdk/blob/d4126181dfc96bcfc3014262326304ed9c2a1401/plugin/key_column_qual.go#L55
* https://github.com/turbot/steampipe-plugin-sdk/blob/49556d376edd992a9ff0cb60485bd50c098604ac/plugin/quals/qual.go#L58


On second thought, this may cause issues if other operators than `=` and `in` are used, e.g. `like` as the string value wouldn't match the checks done afterward. This is because [getQualsValueByColumn` does not consider the operator at all for `string` type](https://github.com/turbot/steampipe-plugin-aws/blob/afb489af301351099084a15df58760e7839351e2/aws/utils.go#L134).

I will do more testing on that.